src/generator/operations.ts

```diff
@@ -289,15 +316,17 @@ export async function generateOperations(session: Session<CodeModel>): Promise<O
     clientText += '\t}\n';
     // propagate optional params
     for (const optionalParam of values(optionalParams)) {
-      if (optionalParam.clientDefaultValue || optionalParam.required === false) {
-        let paramName = optionalParam.language.go!.name;
-        if (isARM) {
-          paramName = `options.${capitalize(optionalParam.language.go!.name)}`;
-        }
-        clientText += `\tif ${paramName} != nil {\n`;
-        clientText += `\t\tclient.${optionalParam.language.go!.name} = *${paramName}\n`;
-        clientText += '\t}\n';
+      if (optionalParam.clientDefaultValue && !isARM) {
+        // in the non-ARM case, these were handled earlier
+        continue;
+      }
+      let paramName = optionalParam.language.go!.name;
+      if (isARM) {
```

Comment:
The `optionalParams` list can contain truly optional or required-with-a-default-value masquerading as optional.  So it could be that `optionalParam.clientDefaultValue` is undefined.